web/frontend/src/ui/RegisterPage.tsx

```diff
@@ -0,0 +1,215 @@
+                                        services.authService.changeLogin(event.target.value)
+                                    }}
+                                    type="login"
+                                    value={login}
+                                    variant="outlined"
+                                />
+                                <TextField
+                                    error={Boolean(touched.password && errors.password)}
+                                    fullWidth
+                                    helperText={touched.password && errors.password}
+                                    label="Password"
+                                    margin="normal"
+                                    name="password"
+                                    onBlur={handleBlur}
+                                    onChange={(event) => {
+                                        handleChange(event)
+                                        services.authService.changePassword(event.target.value)
+                                    }}
+                                    type="password"
+                                    value={password}
+                                    variant="outlined"
+                                />
+                                <Box sx={{ py: 2 }}>
+                                    <Button
+                                        color="primary"
+                                        fullWidth
+                                        size="large"
+                                        type="submit"
+                                        variant="contained"
+                                    >
+                                        Sign up
+                                    </Button>
+                                </Box>
+                                <Typography
+                                    color="textSecondary"
+                                    variant="body1"
+                                >
+                                    Have an account?
+                                    {' '}
+                                    <NavLink to="/login" variant="h6" underline="hover" onClick={() =>{
+                                        services.authService.changePassword('')
+                                        services.authService.changeFirstName('')
+                                        services.authService.changeLastName('')
+                                        services.authService.changeOnline(false)
+                                        services.authService.changeAccessToken('')
+                                    }}>
+                                        Sign in
+                                    </NavLink>
+                                </Typography>
+                            </form>
+                        )}
+                    </Formik>
+                </Container>
+            </Box>
+        </>
```

Comment:
Please split to separate functions.

Reply:
How to split it? Split the container from the form?